common.py

```diff
@@ -759,43 +762,142 @@ def read_training_features(model_file):
     for i in range(len(RHO_edges)-1):
-       for j in range(len(PT_edges)-1):
-          CUT = (rho[cuts]>RHO_edges[i]) & (rho[cuts]<RHO_edges[i+1]) & (pt[cuts]>PT_edges[j]) & (pt[cuts]<PT_edges[j+1])
-          if len(VAR[CUT])==0: continue
-          if len(VAR[CUT])>0:
-             VAR_map[i][j]=np.percentile(VAR[CUT],100-percent) # percent is calculated based on the bdt working point
+        for j in range(len(PT_edges)-1):
+            # Apply cuts to select data in the current rho and pt bin
+            CUT = (rho[cuts]>RHO_edges[i]) & (rho[cuts]<RHO_edges[i+1]) & (pt[cuts]>PT_edges[j]) & (pt[cuts]<PT_edges[j+1])
+
+            # If there is no data in this bin, skip it
+            if len(VAR[CUT])==0: continue
+
+            # If there is data in this bin, calculate the percentile of the variable
+            if len(VAR[CUT])>0:
+                VAR_map[i][j]=np.percentile(VAR[CUT],100-percent) # percent is calculated based on the bdt working point
 
+    # Smooth the variable map using a Gaussian filter
     VAR_map_smooth = gaussian_filter(VAR_map,1)
+
+    # Return the smoothed variable map, along with the rho and pt edges
     return VAR_map_smooth, RHO_edges, PT_edges
 
-def ddt(mt, pt, rho, var, weight, percent):
+# Class that converts numpy arrays into list so they can be easily stored in json files
+class NumpyArrayEncoder(json.JSONEncoder):
+    """Custom encoder for numpy data types."""
+    def default(self, obj):
+        if isinstance(obj, np.ndarray):
+            return obj.tolist()
+        return super(NumpyArrayEncoder, self).default(obj)
+
+def create_DDT_map_dict(mt, pt, rho, var, weight, percents, cut_vals, ddt_name):
+    '''
+    This function creates the dictionary of DDT 2D maps for a range of
+        cut_vals at corresponding bkg efficiencies given as percents
+    The DDT 2D map at each cut_val contains: var_map_smooth, RHO_edges, and PT_edges, 
+        The inputs to the function are (mt, pt, rho, var, weight, percents, cut_vals, ddt_name).
+    In essense the DDT is a 2D function of rho and pt that
+        is calculated using the background data for the variable
+        that you want to decorrelate
+    '''
+
+    # Apply the rho-ddt window cuts to the data
     cuts = rhoddt_windowcuts(mt, pt, rho)
-    var_map_smooth, RHO_edges, PT_edges = varmap(mt, pt, rho, var, weight, percent)
+
+    # Initialize the dictionary
+    var_dict = {}
+
+    # Loop over the values in cut_vals and percents
+    for cut_val, percent in zip(cut_vals, percents):
+
+        print("Creating DDT 2D map for a cut value of ", cut_val, " and an efficiency (percent) of ", percent)
+
+        # Generate a smoothed variable map, along with the rho and pt edges
+        var_map_smooth, RHO_edges, PT_edges = varmap(mt, pt, rho, var, weight, percent)
+
+        # Store the results in the dictionary
+        var_dict[str(cut_val)] = (var_map_smooth, RHO_edges, PT_edges)
+
+    # Save the dictionary to an npz file
```

Comment:
this says npz, but it's actually saved as json. is there a reason for that?

Reply:
Yes, I had trouble coming up with a smart way of storing the values in an npz in a clever way such that they aligned with the bdt cuts. Then I thought to use a json, since the bdt is also already a json, but I forgot to change the comment.